client/signup/config/steps-pure.js

```diff
@@ -919,6 +920,15 @@ export function generateSteps( {
 			providesDependencies: [ 'cartItem' ],
 		},
 
+		'storage-addon': {
+			stepName: 'storage-addon',
+			apiRequestFunction: addAddOnsToCart,
+			fulfilledStepCallback: maybeAddStorageAddonToCart,
```

Comment:
`Non-blocking:`

Would you mind elaborating on `fulfilledStepCallback`? Had some trouble finding docs. I'm assuming it's invoked when `goToNextStep` or `submitSignupStep` is called?

Reply:
There's no documentation yet, unfortunately. The fulfilled step is handled [here](https://github.com/Automattic/wp-calypso/blob/02b0d00bb803e69bd1e4a09631acc0e68a8f645f/client/signup/main.jsx#L180). `fulfilledStepCallback` is used to remove a step from the signup flow if the dependency provided by that step is already provided in a previous step - for example, in the `/start/premium` signup flow, the plans step is hidden because the plan cart item is already provided at the start of the flow (the Premium plan is auto-added to cart).

Reply:
> is used to remove a step from the signup flow if the dependency provided by that step is already provided in a previous step - for example, in the /start/premium signup flow

Thanks for explaining -- much appreciated 🙏